lib/ddtrace/contrib/cucumber/formatter.rb

```diff
@@ -0,0 +1,104 @@
+require 'ddtrace/ext/app_types'
+require 'ddtrace/ext/ci'
+require 'ddtrace/ext/test'
+require 'ddtrace/contrib/analytics'
+require 'ddtrace/contrib/cucumber/ext'
+
+module Datadog
+  module Contrib
+    module Cucumber
+      # Defines collection of instrumented Cucumber events
+      class Formatter
+        attr_reader :config
+        private :config
+
+        attr_reader :current_feature_span, :current_step_span
+        private :current_feature_span, :current_step_span
+
+        def initialize(config)
+          @config = config
+
+          bind_events(config)
+        end
+
+        def bind_events(config)
+          config.on_event :test_case_started, &method(:on_test_case_started)
+          config.on_event :test_case_finished, &method(:on_test_case_finished)
+          config.on_event :test_step_started, &method(:on_test_step_started)
+          config.on_event :test_step_finished, &method(:on_test_step_finished)
+        end
+
+        def on_test_case_started(event)
+          trace_options = {
+            app: Ext::APP,
+            resource: event.test_case.name,
+            service: configuration[:service_name],
+            span_type: Datadog::Ext::AppTypes::TEST,
+            tags: tags.merge(Datadog.configuration.tags)
+          }
+          @current_feature_span = tracer.trace(Datadog::Ext::AppTypes::TEST, trace_options)
+          @current_feature_span.set_tag(Datadog::Ext::Test::FRAMEWORK, Ext::FRAMEWORK)
```

Comment:
We have an unwritten convention to prefix tag constants with `TAG_`:
```suggestion
          @current_feature_span.set_tag(Datadog::Ext::Test::TAG_FRAMEWORK, Ext::FRAMEWORK)
```

Same goes for the other tags below.